src/nvim/eval.c

```diff
@@ -19753,16 +19814,52 @@ char_u *do_string_sub(char_u *str, char_u *pat, char_u *sub, char_u *flags)
   return ret;
 }
 
+static inline JobOptions common_job_options(char **argv, char *autocmd_file)
+{
+  TerminalJobData *data = xcalloc(1, sizeof(TerminalJobData));
+  if (autocmd_file) {
+    data->autocmd_file = xstrdup(autocmd_file);
+  }
+  JobOptions opts = JOB_OPTIONS_INIT;
+  opts.argv = argv;
+  opts.data = data;
+  opts.stdout_cb = on_job_stdout;
+  opts.stderr_cb = on_job_stderr;
+  opts.exit_cb = on_job_exit;
+  return opts;
+}
+
+static inline Job *common_job_start(JobOptions opts, typval_T *rettv)
+{
+  TerminalJobData *data = opts.data;
+  data->refcount++;
+  Job *job = job_start(opts, &rettv->vval.v_number);
+
+  if (rettv->vval.v_number <= 0) {
+    if (rettv->vval.v_number == 0) {
+      EMSG(_(e_jobtblfull));
+      free(opts.term_name);
+      free(data->autocmd_file);
+      free(data);
+    } else {
+      EMSG(_(e_jobexe));
+    }
+    return NULL;
+  }
+  data->job = job;
+  return job;
+}
+
 // JobActivity autocommands will execute vimscript code, so it must be executed
 // on Nvim main loop
-static inline void push_job_event(Job *job, RStream *rstream, char *type)
+static inline void push_job_event(Job *job, char *type, char *data,
+    size_t count)
```

Comment:
> but I'd rather leave this like this for now since its not specified anywhere in our code style that we need to do that

Sure, I don't care that much, just a suggestion.

> and its more painful changing arguments when they need to be aligned.

What do you mean? I use `autoindent` (no `smartindent`) and whenever I type this

```c
int foo(int n1,
```

and hit return, my cursor is lined up with the start of the previous arg,so I can type out the remaining args easily:

```
int foo(int n1,
        int n2,
        int n3,
```